### DISCUSSION TITLE
use-viewport-size should only return valid values, and refrain from sending `0`

### DISCUSSION THREAD

**Discussion by Morriz** [mantinedev/mantine#4727] [Feature requests]
### What package has an issue

@mantine/core

### Describe the bug

use-viewport-size is triggering unnecessary because the package emits a value of `0` as the docs say. But why it is not silenced eludes me? If there is a need to export another prop like `isRotating` then please do so, but I see no use case for the current behavior.

### What version of @mantine/hooks page do you have in package.json?

v6.0.13

### Do you know how to fix the issue

None

### Are you willing to participate in fixing this issue and create a pull request with the fix

None

**Comment by rtivital** [comment]
use-viewport-size hook cannot get the viewport value on the server. If it cannot retrieve the value, it will return 0.

**Comment by Morriz** [comment]
Since setters take a tick, we can improve on what I see now. How about something like this change:

from:
```ts
export function useViewportSize() {
  const [windowSize, setWindowSize] = useState({
    width: 0,
    height: 0,
  });

  const setSize = useCallback(() => {
    setWindowSize({ width: window.innerWidth || 0, height: window.innerHeight || 0 });
  }, []);

  useWindowEvent('resize', setSize, eventListerOptions);
  useWindowEvent('orientationchange', setSize, eventListerOptions);
  useEffect(setSize, []);

  return windowSize;
}
```

to this:

```ts
export function useViewportSize() {
  const [windowSize, setWindowSize] = useState({
    width: window?.innerWidth ?? 0, // <-- we set it because we can and thus avoids the glitch
    height: window?.innerHeight ?? 0,
  });

  const setSize = useCallback(() => {
    setWindowSize({ width: window!.innerWidth ?? 0, height: window!.innerHeight ?? 0 });
  }, []);

  useWindowEvent('resize', setSize, eventListerOptions);
  useWindowEvent('orientationchange', setSize, eventListerOptions);
  useEffect(setSize, []);

  return windowSize;
}
```

**Comment by Morriz** [comment]
I did that just now and it works fine. I will make a PR if you care...

**Comment by Morriz** [comment]
I also made some slight improvements in the typescript according to best practices ;)

**Comment by rtivital** [comment]
>     width: window?.innerWidth ?? 0, // <-- we set it because we can and thus avoids the glitch

This will produce a hydration mismatch if the value is used anywhere in styles or website content. In other words, the hook will not work with Next.js, Remix and other ssr environments.

**Comment by Morriz** [comment]
Maybe make it configurable by passing a prop then? Those using SSR may deal with the glitch ;p
